token-metadata/program/src/processor/metadata/update.rs

```diff
@@ -137,11 +138,19 @@ fn update_v1(program_id: &Pubkey, ctx: Context<Update>, args: UpdateArgs) -> Pro
         token_account: token.as_ref(),
         metadata_delegate_record_info: ctx.accounts.delegate_record_info,
         metadata_delegate_role: Some(MetadataDelegateRole::Update),
+        precedence: &[
+            AuthorityType::Metadata,
+            AuthorityType::Holder,
```

Comment:
Does it make sense to have the `Holder` here? What if the holder happens to have a delegate, then this will always evaluate to `Holder`, which might not be desirable.

Reply:
Good point. Also, only `Metadata` authority type is supported right now so I could just pass that in but for future proofing it I think `Metadata` and `Delegate` are probably sufficient?

Reply:
Yes, agreed.